druid-shell/src/util.rs

```diff
@@ -32,6 +32,7 @@ fn current_thread_id() -> u64 {
 /// # Panics
 ///
 /// Panics when called from a non-main thread.
+#[allow(dead_code)] // can be unused on some backend
```

Comment:
I'd prefer to use either use [`cfg_atr`](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg_attr-attribute) to turn this on per-platform, or use `cfg` to only compile this on platforms where it's used.

(I don't like blanket `allow`s like this because you can't tell from lints when something is actually not used anymore)